### DISCUSSION TITLE
Advantages of Using Symbols in Material Components like 'get propName' Example?

### DISCUSSION THREAD

**Discussion by ChellappanRajan** [material-components/material-web#5068] [General]
I've noticed that in many Material Components, such as the example in this code (https://github.com/material-components/material-web/blob/main/radio/internal/radio.ts#L19), symbols are used instead of 'propName.' Could you please explain the advantages of using symbols in this context?

**Comment by asyncliz** [comment]
We usually use them for one of two use cases:

1. To expose access for things that need to be publicly used outside of the class, but not exposed to users (such as [`ElementInternals` access for shared code](https://github.com/material-components/material-web/blob/04dcf25c4ae6bb334ed106b5f73ee1104e50d17c/internal/controller/element-internals.ts#L24))
2. To use as private properties without creating additional members, such as `this.checked` and `this._checked`.
